src/deltablocks.cpp

```diff
@@ -558,6 +558,9 @@ bool CheckBobtailPoW(CBlockHeader deltaHeader, std::vector<uint256> ancestors, c
     bool fOverflow;
     arith_uint256 bnTarget;
 
+    if (ancestors.size() < (uint8_t)(k-1))
```

Comment:
there is no need to recast k to uint8_t it is already of that type. 
there should be a check before this comparison to check if k == 0 to avoid evaluating 0 - 1 which will result in 255 for a uint8_t